cross-modal-search/flows/flow-index.yml

```diff
@@ -1,40 +1,41 @@
-version: '1'
-with:
-  prefetch: 10
-  port_expose: 45678
-  workspace: $JINA_WORKSPACE
-pods:
-  - name: loader                                      # load images from the dataset of image-caption pairs
-    uses: pods/image-load.yml
-    shards: $JINA_PARALLEL
-    read_only: true
-    needs: [gateway]
-  - name: image_encoder                               # encode images into embeddings with CLIP model
-    uses: 'jinahub://CLIPImageEncoder'
-    volumes: $HOME/.cache/clip:/root/.cache/clip
-    shards: $JINA_PARALLEL
-    timeout_ready: 600000
-    read_only: true
-  - name: image_vector_indexer                        # store image embeddings
-    polling: any
-    uses: pods/index-image-vector.yml
-    shards: $JINA_SHARDS
-  - name: image_kv_indexer                            # store image documents
-    polling: any
-    uses: pods/index-image-kv.yml
-    shards: $JINA_SHARDS
-    needs: [gateway]
-  - name: text_encoder                                # encode text into embeddings with CLIP model
-    uses: pods/clip/text-encoder.yml
-    shards: $JINA_PARALLEL
-    timeout_ready: 600000
-    read_only: true
-    needs: [gateway]
-  - name: text_indexer                                # index the text into documents
-    polling: any
-    uses: pods/index-text.yml #(numpy + binary pb indexer)
-    shards: $JINA_SHARDS
-    needs: text_encoder
-  - name: join_all                                    # wait on the 3 executors to finish data processing with "needs"
-    needs: [image_vector_indexer, image_kv_indexer, text_indexer]
+jtype: Flow                                       # We configure the index flow here that is used for indexing images and captions
+version: '1'                                      # yml version
+with:                                             # Parameters for the flow are defined after with
+  prefetch: 10                                    # Number of prefetched requests from the client
+  port_expose: $JINA_PORT                         # Port defined in environment variable
+  workspace: $JINA_WORKSPACE                      # Workspace folder
+pods:                                             # Now, we define the pods that are used
+  - name: image_loader                            # The first executor is an image loader that filters only image documents
+    uses: ImageReader                             # Type of the executor
+    py_modules: 'flows/executors.py'              # The python file where the executor is implemented
+    read_only: true                               # Executor does not modify files
+    needs: gateway                                # Executor is after the gateway, this means at the start of the flow
+  - name: image_encoder                           # After the images are read, compute their embedding in the encoder
+    uses: 'jinahub+docker://CLIPImageEncoder'     # The type of the executor - here, we use a hub executor from the jinahub in the form of a docker container
```

Comment:
If we want to retain docker flow, this may introduce a bug as we would run into a dokcer-in-docker situation, which is currently not supported.

Reply:
You could try running docker-from-docker - by bind-mounting the docker socket into the flow container.

Reply:
I have removed all the docker containers and Dockerfiles for the examples. Therefore, this should be fine.